src/cloudflare/internal/ai-api.ts

```diff
@@ -98,9 +104,18 @@ export class Ai {
     /*
      * @deprecated this method is deprecated, do not use this
      */
-    public getLogs(): Array<string> {
+    public getLogs(): string[] {
       return []
     }
+
+    private async _parseError(res: Response): Promise<InferenceUpstreamError> {
+      try {
+        const content = (await res.json()) as AiError
+        return new InferenceUpstreamError(`${content.internalCode}: ${content.description}`, content.name);
+      } catch {
+        return new InferenceUpstreamError(await res.text());
```

Comment:
Hey :wave:,

@Cherry sent me this PR, and while browsing through I noticed that this wouldn't actually work. Even if the JSON parsing fails, such as on an empty response body, the body would be already consumed and this would throw again.

Reply:
Thanks for pointing that out, i've fixed the issue and added a unit test to cover it